spark-cassandra-connector/src/main/scala/com/datastax/spark/connector/rdd/CassandraRDD.scala

```diff
@@ -276,7 +276,37 @@ class CassandraRDD[R] private[connector] (
   lazy val tableDef = {
     Schema.fromCassandra(connector, Some(keyspaceName), Some(tableName)).tables.headOption match {
       case Some(t) => t
-      case None => throw new IOException(s"Table not found: $keyspaceName.$tableName")
+      case None => {
+        val clusterMetadata = connector.withClusterDo(_.getMetadata)
+        val keyspaces = clusterMetadata.getKeyspaces
+        val ksMap = keyspaces.groupBy(_.getName.toLowerCase).toMap
+        ksMap.get(keyspaceName.toLowerCase) match {
+          case Some(keyspaceList) => {
+            val possibleTables =
+              keyspaceList.flatMap(_.getTables)
+                .filter(_.getName.toLowerCase == tableName.toLowerCase)
+                .map(table => table.getKeyspace.getName + "." + table.getName)
+            if (!possibleTables.isEmpty) {
+              val errorString = possibleTables.mkString("\n")
+              throw new IOException(
+                s"""Couldn't find: $keyspaceName.$tableName
+                    |Other tables exist with different cases
+                    |Did you mean :
+                    |$errorString""".stripMargin)
+            } else {
+              val errorString = keyspaceList.map(_.getName).mkString("\n")
+              throw new IOException(
+                s"""Couldtn't find: $keyspaceName.$tableName
+                    |Other keyspaces exist with different cases
+                    |Did you mean:
+                    |$errorString""".stripMargin)
+            }
+          }
+          case None => {
+            throw new IOException(s"Table not found: $keyspaceName.$tableName")
+          }
+        }
+      }
     }
```

Comment:
Abstract this logic out to the Schema object or add methods for obtaining a list of similar keyspaces / tables, etc. and just call it from here. This is because this code is not really related to RDD.